src/vs/workbench/electron-main/main.ts

```diff
@@ -253,7 +255,11 @@ function setupIPC(): TPromise<Server> {
 // and assign them to the process environment (e.g. when doubleclick app on Mac)
 getUserEnvironment()
 	.then(userEnv => {
-		assign(process.env, userEnv);
+		if (!tty.isatty(1)) {
+			// Only assign global environment variables when
+			// STDOUT is not attached to terminal.
+			assign(process.env, userEnv);
```

Comment:
https://nodejs.org/api/process.html#process_process_env

> Assigning a property on process.env will implicitly convert the value to a string. 

looks like I better not replace process.env like that unless I also modify userEnv's setter.